File: test_vertex_direct.js
import fs from 'fs/promises';
import { GoogleAuth } from 'google-auth-library';

// Configuration
const CONFIG = {
  projectId: "456295042668",
  location: "us-central1",
  endpointId: "8159951878260523008"
};

// Load a test image
async function loadTestImage() {
  try {
    // Try to load the test image from the test_payload.json file
    const payload = JSON.parse(await fs.readFile('./test_payload.json', 'utf8'));
    return payload.instances[0].content;
  } catch (error) {
    console.error('Error loading test image:', error);
    // Return a minimal base64 image if we can't load the test image
    return '/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDAAMCAgICAgMCAgIDAwMDBAYEBAQEBAgGBgUGCQgKCgkICQkKDA8MCgsOCwkJDRENDg8QEBEQCgwSExIQEw8QEBD/2wBDAQMDAwQDBAgEBAgQCwkLEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBD/wAARCAABAAEDASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwD9U6KKKACiiigAooooA//Z';
  }
}

// Get an access token
async function getAccessToken() {
  const auth = new GoogleAuth({
    scopes: ['https://www.googleapis.com/auth/cloud-platform']
  });
  
  const client = await auth.getClient();
  const token = await client.getAccessToken();
  return token.token;
}

// Test different formats
async function testVertexDirect() {
  try {
    console.log('Testing direct Vertex AI call...');
    
    // Get an access token
    const accessToken = await getAccessToken();
    console.log('Got access token');
    
    // Load the test image
    const imageContent = await loadTestImage();
    console.log(`Loaded test image (${imageContent.length} characters)`);
    
    // Create the request payload with the correct format
    const payload = {
      instances: [
        {
          content: imageContent,
          mimeType: "image/jpeg"
        }
      ],
      parameters: {
        confidenceThreshold: 0.0,
        maxPredictions: 5
      }
    };
    
    // Make the request to the Vertex AI endpoint
    console.log('Sending request to Vertex AI endpoint...');
    const vertexEndpoint = `https://${CONFIG.location}-aiplatform.googleapis.com/v1/projects/${CONFIG.projectId}/locations/${CONFIG.location}/endpoints/${CONFIG.endpointId}:predict`;
    
    const response = await fetch(vertexEndpoint, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(payload)
    });
    
    // Check the response
    const statusCode = response.status;
    console.log(`Vertex AI responded with status code: ${statusCode}`);
    
    let responseData;
    try {
      responseData = await response.json();
    } catch (error) {
      const text = await response.text();
      console.error('Error parsing response as JSON:', error);
      console.log('Response text:', text);
      return;
    }
    
    // Log the response
    if (statusCode === 200) {
      console.log('Vertex AI response:', JSON.stringify(responseData, null, 2));
    } else {
      console.error('Vertex AI error:', responseData);
    }
  } catch (error) {
    console.error('Test failed:', error);
  }
}

// Run the test
testVertexDirect(); 

File: test_vertex_direct.py
#!/usr/bin/env python3

import json
import base64
import requests
import subprocess
import os
from PIL import Image
import io

# Configuration
PROJECT_ID = "456295042668" 
LOCATION = "us-central1"
MODEL_ID = "6041241350047793152"  # Model ID (not endpoint ID)
TEST_IMAGE_PATH = "/Users/gabe/VEXUS/test_image.png"

print("========== VERTEX AI DIRECT MODEL TEST ==========")

def get_access_token():
    """Get access token from gcloud"""
    print("Getting access token...")
    result = subprocess.run(
        ["gcloud", "auth", "print-access-token"], 
        capture_output=True, 
        text=True, 
        check=True
    )
    token = result.stdout.strip()
    print("✓ Access token obtained successfully")
    return token

# Load the test image
try:
    with open(TEST_IMAGE_PATH, "rb") as image_file:
        image_data = image_file.read()
        image_size = len(image_data)
        print(f"Loaded test image, size: {image_size} bytes")

        # Check the actual image type regardless of extension
        img = Image.open(io.BytesIO(image_data))
        mime_type = f"image/{img.format.lower()}"
        print(f"Image format detected: {img.format} (MIME type: {mime_type})")

        # Convert to base64
        base64_data = base64.b64encode(image_data).decode("utf-8")
except FileNotFoundError:
    print(f"Test image not found at {TEST_IMAGE_PATH}. Using fallback.")
    # Create a simple 1x1 pixel PNG as fallback
    img = Image.new('RGB', (1, 1), color = 'white')
    buffered = io.BytesIO()
    img.save(buffered, format="PNG")
    base64_data = base64.b64encode(buffered.getvalue()).decode("utf-8")
    mime_type = "image/png"
    print("Created fallback 1x1 pixel image")

# Get access token
token = get_access_token()

# Try direct model prediction (not endpoint)
print("\nMaking direct Vertex AI model prediction (bypassing endpoint)...")
model_url = f"https://{LOCATION}-aiplatform.googleapis.com/v1/projects/{PROJECT_ID}/locations/{LOCATION}/models/{MODEL_ID}:predict"

# Different request formats to try
formats_to_try = [
    # Format 1: Standard format with mimeType
    {
        "instances": [
            {
                "content": base64_data,
                "mimeType": mime_type
            }
        ]
    },
    # Format 2: No mimeType
    {
        "instances": [
            {
                "content": base64_data
            }
        ]
    },
    # Format 3: Raw base64 in array format
    {
        "instances": [[base64_data]]
    },
    # Format 4: Content in array format
    {
        "instances": [[{"b64": base64_data}]]
    }
]

# Try each format
for i, payload in enumerate(formats_to_try):
    print(f"\nTrying request format {i+1}...")
    
    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json"
    }
    
    try:
        response = requests.post(model_url, headers=headers, json=payload)
        
        print(f"Status code: {response.status_code}")
        
        if response.status_code == 200:
            result = response.json()
            print("✓ Prediction successful!")
            print("Response:")
            print(json.dumps(result, indent=2))
            print(f"\nFormat {i+1} worked!")
            break
        else:
            print("✗ Prediction failed")
            try:
                error = response.json()
                print(f"Error details: {json.dumps(error, indent=2)}")
            except:
                print(f"Error response: {response.text}")
    except Exception as e:
        print(f"Request error: {e}")

print("\n========== TEST COMPLETE ==========") 